File: editor/src/main/java/org/test/editor/core/service/FolderService.java
package org.test.editor.core.service;

import org.test.editor.core.dto.CreateFolderDTO;
import org.test.editor.core.dto.CreatePrimaryFolderDTO;
import org.test.editor.core.dto.FolderDTO;
import org.test.editor.core.model.Folder;

import java.util.Optional;

public interface FolderService {
    FolderDTO createPrimaryFolder(CreatePrimaryFolderDTO createPrimaryFolderDTO);
    FolderDTO createFolder(CreateFolderDTO createFolderDTO);
    Optional<Folder> getFolderById(Integer folderId);
    void delete(Integer folderId);
}

File: code-editor/src/app/core/services/folder.service.ts
import { Injectable } from '@angular/core';
import { EditorOperationsWebsocketService } from './editor-operations-websocket.service';
import { CreateFolderDTO } from '../dto/create-folder.dto';
import { CreateFolderPrimaryDTO } from '../dto/create-folder-primary.dto';


@Injectable({
  providedIn: 'root'
})
export class FolderService {
    constructor(private webSocketService: EditorOperationsWebsocketService) { }

    createFolder(createFolderDTO: CreateFolderDTO): void {
      this.webSocketService.sendMessage('create-folder', createFolderDTO);
    }
    createFolderPrimary(createFolderPrimaryDTO: CreateFolderPrimaryDTO): void {
      this.webSocketService.sendMessage('create-primary-folder', createFolderPrimaryDTO);
    }
  
    deleteFolder(folderId: number): void {
      this.webSocketService.sendMessage('delete-folder', folderId);
    }
}